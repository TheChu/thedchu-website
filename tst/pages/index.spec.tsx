import { render } from '@testing-library/react';
import { useStaticQuery } from 'gatsby';
import React from 'react';
import uuid from 'short-uuid';
import IndexPage from '../../src/pages/index';

jest.mock('short-uuid');
(uuid.generate as jest.Mock).mockReturnValue('test-uuid');

(useStaticQuery as jest.Mock).mockReturnValue({
  allMarkdownRemark: {
    edges: [
      {
        node: {
          frontmatter: {
            title: 'Test Title',
            date: 'June 200X',
          },
          thumbnail: {
            childImageSharp: {
              gatsbyImageData: {
                layout: 'constrained',
                placeholder: {
                  fallback: 'test-base64',
                },
                images: {
                  fallback: {
                    src: 'test-src',
                    srcSet: 'test-srcSet',
                    sizes: '100vw',
                  },
                  sources: [],
                },
                width: 1,
                height: 0.6666666666666666,
              },
              original: { src: 'test-src' },
            },
          },
        },
      },
    ],
  },
  mastheadImage: {
    localImage: {
      childImageSharp: {
        gatsbyImageData: {
          layout: 'fullWidth',
          placeholder: {
            fallback: 'test-base64',
          },
          images: {
            fallback: {
              src: 'test-src',
              srcSet: 'test-srcSet',
              sizes: '100vw',
            },
            sources: [],
          },
          width: 1,
          height: 0.6666666666666666,
        },
      },
    },
  },
  site: { siteMetadata: { title: 'thedchu' } },
});

describe('IndexPage', () => {
  it('should match IndexPage snapshot', () => {
    const { asFragment } = render(<IndexPage />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render navigation bar', () => {
    const { getByRole } = render(<IndexPage />);
    expect(getByRole('navigation')).toBeVisible();
  });

  it('should render `Photos` section', () => {
    const { getByRole } = render(<IndexPage />);
    expect(getByRole('region', { name: 'Photos' })).toBeVisible();
  });

  it('should render footer', () => {
    const { getByRole } = render(<IndexPage />);
    expect(getByRole('contentinfo')).toBeVisible();
  });
});
