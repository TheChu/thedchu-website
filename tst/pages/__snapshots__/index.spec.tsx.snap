// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage should match IndexPage snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-light fixed-top py-3 "
    id="mainNav"
  >
    <div
      class="container"
    >
      <div
        class="navbar-item"
      >
        <ul
          class="navbar-nav mr-auto my-2 my-lg-0"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link nav-link-icon mx-2"
              href="https://instagram.com/thedchu"
            >
              <i
                aria-label="Instagram"
                class="fab fa-instagram"
              />
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link nav-link-icon mx-2"
              href="https://linkedin.com/in/thedchu"
            >
              <i
                aria-label="LinkedIn"
                class="fab fa-linkedin-in"
              />
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link nav-link-icon mx-2"
              href="https://github.com/thechu"
            >
              <i
                aria-label="Github"
                class="fab fa-github"
              />
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link nav-link-icon mx-2"
              href="mailto:chu.david.a@gmail.com"
            >
              <i
                aria-label="Email"
                class="fas fa-envelope"
              />
            </a>
          </li>
        </ul>
      </div>
      <a
        class="navbar-brand navbar-item"
        href="/"
      >
        <svg
          aria-label="DChu"
          class="navbar-brand-logo"
          viewBox="0 0 1000 512"
        >
          <use
            href="test-file-stub#thedchu-logo"
          />
        </svg>
      </a>
      <div
        class="navbar-toggler-container"
      >
        <button
          aria-controls="navbarResponsive"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler navbar-toggler-right collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
      </div>
      <div
        class="collapse navbar-collapse "
        id="navbarResponsive"
      >
        <ul
          class="navbar-nav ml-auto my-2 my-lg-0"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/about"
            >
              About
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="/"
            >
              Photos
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
  <header
    class="masthead gbi--1668593287-test-uuid"
    id="top"
    style="background-position: center; background-repeat: no-repeat; background-size: cover; position: relative; opacity: 0.99;"
  >
    <style>
      
          .masthead.gbi--1668593287-test-uuid:before,
          .masthead.gbi--1668593287-test-uuid:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            background-color: #040e18;
            transition: opacity 0.5s ease 250ms;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .masthead.gbi--1668593287-test-uuid:before {
            z-index: -100;
            background-image: url(test-base64);
            
            opacity: 1; 
          }
          .masthead.gbi--1668593287-test-uuid:after {
            z-index: -101;
            
            background-image: url(test-base64);
            
          }
        
    </style>
    <div
      class="container h-100"
    >
      <div
        class="row h-100 align-items-center justify-content-center text-center"
      >
        <div
          class="col-lg-10 align-self-end"
        >
          <h1
            class="text-uppercase text-white font-weight-bold"
          >
            <svg
              aria-label="DChu"
              class="masthead-title"
              viewBox="0 0 1000 512"
            >
              <use
                href="test-file-stub#thedchu-logo"
              />
            </svg>
          </h1>
        </div>
        <div
          class="col-lg-8 align-self-baseline"
        >
          <p
            class="text-light font-weight-light mb-5"
          >
            hello my name dchu I like to take nice photos, here are some
          </p>
          <a
            class="btn btn-primary btn-xl js-scroll-trigger"
            href="/#photos"
          >
            Find Out More
          </a>
        </div>
      </div>
    </div>
  </header>
  <section
    aria-label="Photos"
    class="bg-dark"
    id="photos"
  >
    <div
      class="container-fluid p-0"
    >
      <div
        class="row no-gutters"
      >
        <div
          class="col-lg-3 col-sm-6"
        >
          <a
            aria-label="Test Title"
            class="photos-box"
            href="test-src"
          >
            <div
              class="gatsby-image-wrapper gatsby-image-wrapper-constrained"
              data-gatsby-image-wrapper=""
              style="position: relative; overflow: hidden; display: inline-block; vertical-align: top;"
            >
              <div
                style="max-width: 1px; display: block;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;charset=utf-8,%3Csvg height='0.6666666666666666' width='1' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
                  style="max-width: 100%; display: block; position: static;"
                />
              </div>
              <img
                alt=""
                aria-hidden="true"
                data-placeholder-image=""
                decoding="async"
                src="test-base64"
                style="height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
              />
              <img
                alt="Test Title"
                data-gatsby-image-ssr=""
                data-main-image=""
                data-src="test-src"
                data-srcset="test-srcSet"
                decoding="async"
                loading="lazy"
                sizes="100vw"
                style="height: 100%; left: 0px; position: absolute; top: 0px; transform: translateZ(0); transition: opacity 250ms linear; width: 100%; will-change: opacity; opacity: 0;"
              />
              <noscript />
              <script
                type="module"
              >
                const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1)}}
              </script>
            </div>
            <div
              class="photos-box-caption"
            >
              <div
                class="project-category text-white-50"
              >
                June 200X
              </div>
              <div
                class="project-name"
              >
                Test Title
              </div>
            </div>
          </a>
        </div>
      </div>
    </div>
  </section>
  <footer
    class="bg-dark py-5"
  >
    <div
      class="container"
    >
      <div
        class="row justify-content-center"
      >
        <div
          class="col-lg-8 text-center mt-5 mb-lg-0"
        >
          <a
            class="mx-2"
            href="https://instagram.com/thedchu"
          >
            <i
              aria-label="Instagram"
              class="fab fa-3x mb-3 fa-instagram"
            />
          </a>
          <a
            class="mx-2"
            href="https://linkedin.com/in/thedchu"
          >
            <i
              aria-label="LinkedIn"
              class="fab fa-3x mb-3 fa-linkedin-in"
            />
          </a>
          <a
            class="mx-2"
            href="https://github.com/thechu"
          >
            <i
              aria-label="Github"
              class="fab fa-3x mb-3 fa-github"
            />
          </a>
          <a
            class="mx-2"
            href="mailto:chu.david.a@gmail.com"
          >
            <i
              aria-label="Email"
              class="fas fa-3x mb-3 fa-envelope"
            />
          </a>
        </div>
      </div>
      <div
        class="small text-center text-muted"
      >
        Copyright Â© 2021 - David Chu
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
